store/postgres/src/mapping.rs

```diff
@@ -0,0 +1,1141 @@
+            });
+            changes.extend(set);
+        }
+        Ok((StoreEvent::new(changes), count))
+    }
+}
+
+impl AsDdl for Mapping {
+    fn fmt(&self, f: &mut dyn fmt::Write, mapping: &Mapping) -> fmt::Result {
+        // We sort tables here solely because the unit tests rely on
+        // 'create table' statements appearing in a fixed order
+        let mut tables = self.tables.values().collect::<Vec<_>>();
+        tables.sort_by(|a, b| (&a.position).partial_cmp(&b.position).unwrap());
+        // Output 'create table' statements for all tables
+        for table in tables {
+            table.fmt(f, mapping)?;
+        }
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Reference {
+    pub table: SqlName,
+    pub column: SqlName,
+}
+
+impl Reference {
+    fn to_column(table: SqlName, column: &Column) -> Reference {
+        let column = match &column.derived {
+            Some(col) => col,
+            None => PRIMARY_KEY_COLUMN,
+        }
+        .into();
+
+        Reference { table, column }
+    }
+}
+
+/// This is almost the same as graph::data::store::ValueType, but without
+/// ID and List; with this type, we only care about scalar types that directly
+/// correspond to Postgres scalar types
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ColumnType {
+    Boolean,
+    BigDecimal,
+    BigInt,
+    Bytes,
+    Int,
+    String,
+}
+
+impl From<IdType> for ColumnType {
+    fn from(id_type: IdType) -> Self {
+        match id_type {
+            IdType::Bytes => ColumnType::Bytes,
+            IdType::String => ColumnType::String,
+        }
+    }
+}
+
+impl ColumnType {
+    fn from_value_type(value_type: ValueType, id_type: IdType) -> Result<ColumnType, StoreError> {
+        match value_type {
+            ValueType::Boolean => Ok(ColumnType::Boolean),
+            ValueType::BigDecimal => Ok(ColumnType::BigDecimal),
+            ValueType::BigInt => Ok(ColumnType::BigInt),
+            ValueType::Bytes => Ok(ColumnType::Bytes),
+            ValueType::Int => Ok(ColumnType::Int),
+            ValueType::String => Ok(ColumnType::String),
+            ValueType::ID => Ok(ColumnType::from(id_type)),
+            ValueType::List => Err(StoreError::Unknown(format_err!(
+                "can not convert ValueType::List to ColumnType"
+            ))),
+        }
+    }
+
+    fn sql_type(&self) -> &str {
+        match self {
+            ColumnType::Boolean => "boolean",
+            ColumnType::BigDecimal => "numeric",
+            ColumnType::BigInt => "numeric",
+            ColumnType::Bytes => "bytea",
+            ColumnType::Int => "integer",
+            ColumnType::String => "text",
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Column {
+    pub name: SqlName,
+    pub field: String,
+    pub field_type: q::Type,
+    pub column_type: ColumnType,
+    pub derived: Option<String>,
+    pub references: Vec<Reference>,
+}
+
+#[allow(dead_code)]
```

Comment:
Lets move this down to the specific dead functions.

Reply:
Turns out, it wasn't necessary at all. Commit 146487f7